### PR TITLE
core: kt-infra: migrate MRSP

### LINKED ISSUE #4263: core: kt-infra: Add a way to generate an `MRSP`
We need to implement a way to generate an `MRSP` from the new path classes.

Depends at least on https://github.com/DGEXSolutions/osrd/issues/4262 and https://github.com/DGEXSolutions/osrd/issues/4261

### PR DESCRIPTION
**Pull request by Erashin**
MRSP renamed to LegacyMRSP. New MRSP class using Path.kt.

Fixes #4263.

### CODE REVIEW

File: core/src/main/java/fr/sncf/osrd/envelope_sim_infra/MRSP.java [hunk 1]
```diff
@@ -1,70 +1,57 @@
...
-                if (Double.isInfinite(speed) || speed == 0)
-                    continue;
+        var offset = addRollingStockLength ? rollingStock.length : 0.;
+        var speedRanges = path.getSpeedRanges(trainTag);
+        for (var speedRange: speedRanges) {
+            // Compute where this limit is active from and to
+            var start = toMeters(speedRange.getLower());
+            var end = Math.min(pathLength, offset + toMeters(speedRange.getUpper()));
+            var speed = toMetersPerSecond(rangeMapEntryToSpeed(speedRange));
+            if (!isInfinite(speed) && speed != 0)
```
Comment (eckter) [inline core/src/main/java/fr/sncf/osrd/envelope_sim_infra/MRSP.java]: In this setting, the speed can't be infinite (can't be represented as long). We *shouldn't* have infinite values here.

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/Path.kt [hunk 2]
```diff
@@ -36,6 +37,8 @@ interface Path {
     fun getCatenary(): DistanceRangeMap<String>
     @JvmName("getDeadSections")
     fun getDeadSections(): DistanceRangeMap<DeadSection>
+    @JvmName("getSpeedRanges")
+    fun getSpeedRanges(trainTag: String?): DistanceRangeMap<Speed>
```
Comment (multun) [inline core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/Path.kt]:
```suggestion
    @JvmName("getSpeedLimits")
    fun getSpeedLimits(trainTag: String?): DistanceRangeMap<Speed>
```

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/PathImpl.kt [hunk 3]
```diff
@@ -53,6 +54,10 @@ data class PathImpl(
         return getRangeMap { dirChunkId -> infra.getTrackChunkDeadSection(dirChunkId) }
     }
 
+    override fun getSpeedRanges(trainTag: String?): DistanceRangeMap<Speed> {
```
Comment (multun) [inline core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/PathImpl.kt]:
```suggestion
    override fun getSpeedLimits(trainTag: String?): DistanceRangeMap<Speed> {
```

**Review by multun — APPROVED** [review]

File: core/src/main/java/fr/sncf/osrd/envelope_sim_infra/MRSP.java [hunk 4]
```diff
@@ -1,70 +1,56 @@
...
-                var end = offset + interval.upperEndpoint();
-                if (addRollingStockLength) {
-                    end += rollingStock.length;
-                    end = Math.min(pathLength, end);
-                }
-                var speed = speedRange.getValue().getSpeedLimit(tag);
-                if (Double.isInfinite(speed) || speed == 0)
-                    continue;
+        var offset = addRollingStockLength ? rollingStock.length : 0.;
+        var speedRanges = path.getSpeedRanges(trainTag);
```
Comment (multun) [inline core/src/main/java/fr/sncf/osrd/envelope_sim_infra/MRSP.java]:
```suggestion
        var speedRanges = path.getSpeedLimits(trainTag);
```

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/Path.kt (hunk 2 above)
Comment (Erashin) [reply core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/Path.kt]: Also changed the corresponding variable names where this method was used.

**Pull request merged**